src/chapters/spring-model-validation/thymeleaf-form-tools.rst

```diff
@@ -0,0 +1,224 @@
+Thymeleaf Form Tools
+====================
+
+Thymeleaf provides some handy attributes for working with form fields. They make use of the fact that, when using model binding, each model field contains a lot of information about the corresponding form input needed. In particular, the model field and its annotations often determine:
+
+#. The name of the form field (that is, the value of its ``name`` attribute).
+#. The validation rules and corresponding error messages for the field.
+
+Thymeleaf provides some convenient attributes to make wiring up a form much easier and cleaner.
+
+Display Validation Errors for a Field - Video
+----------------------------------------------
+
+.. raw:: html
+
+   <div style="text-align:center;"><iframe width="800" height="450" src="https://www.youtube.com/embed/yc-bSDSDuKg" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe></div>
+
+The final code from this video is in the `display-errors branch <https://github.com/LaunchCodeEducation/coding-events/tree/display-errors>`__ of ``coding-events``.
+
+Display Validation Errors for a Field - Text
+--------------------------------------------
+
+Using ``th:field``
+^^^^^^^^^^^^^^^^^^
+
+.. index:: ! th:field
+
+We can use the ``th:field`` attribute on a form input to instruct Thymeleaf to add field-specific attributes to the form, such as ``name`` and ``id``. 
+
+Consider our second input, which currently looks like this:
+
+.. sourcecode:: html
+   :lineno-start: 17
+
+   <label>Description
+      <input type="text" name="description" class="form-control">
+   </label>
+
+We set ``name="description"`` because the field that we want the data in the input to be bound to on submission is the ``description`` field of the ``Event`` class. 
+
+A better approach uses ``th:field`` to bind the ``description`` field *when the form is rendered*.
+
+.. sourcecode:: html
+   :lineno-start: 17
+
+   <label>Description
+      <input type="text" th:field="${event.description}" class="form-control">
+   </label>
+
+With this syntax, Thymeleaf will look for a variable named ``event`` and use its ``description`` property to set the values of the ``name`` and ``id`` attributes. The generated input looks like this:
+
+.. sourcecode:: html
+
+   <input type="text" id="description" name="description" class="form-control">
+
+We don't need to use the ``id`` attribute in this case, but it doesn't hurt anything by being there. Now, Thymeleaf sets ``name`` on its own, and if we were to change the field name on the class, no template updates would be required. 
+
+.. index:: ! no-arg constructor
```

Comment:
i just pushed some edits on the #70 pr (quick-fixes) that include a no-arg index entry and definition in the inheritance chapter so I would remove the "!" from this entry